docs/tools/new_tools_temp.md

```diff
@@ -0,0 +1,186 @@
+- NFC: [https://github.com/ARMmbed/mbed-os-example-nfc](https://github.com/ARMmbed/mbed-os-example-nfc)
+- Sockets: [https://github.com/ARMmbed/mbed-os-example-sockets](https://github.com/ARMmbed/mbed-os-example-sockets)
+
+## Configure the project
+
+The Mbed OS configuration system parses the configuration files in your project (mbed_lib.json, mbed_app.json and targets.json) for a particular target and toolchain, and outputs a CMake script. The build system uses this script to build for your target, using your toolchain.
+
+<span class="tips">**Tip:** If you're rebuilding for the same target and toolchain, you can keep using the same CMake script, so you won't have to use the `configure` command again for each build. If you change your target or toolchain, run the `configure` command again to generate a new CMake script.</span>
+
+1. Check your board's build target name.
+
+    Connect your board over USB and run the `devices` command:
+
+    ```
+    mbed-tools devices
+
+    Board name    Serial number             Serial port             Mount point(s)    Build target(s)
+    ------------  ------------------------  ----------------------  ----------------  -----------------
+    FRDM-K64F     024002017BD34E0F862DB3B7  /dev/tty.usbmodem14402  /Volumes/MBED     K64F
+    ```
+1. To prepare the Mbed configuration information for use with a specific target and toolchain, navigate to the project's root folder and run:
+
+    ```
+    mbed-tools configure -m <target> -t <toolchain>
+    ```
+
+    - The supported targets are `K64F`, `DISCO_L475VG_IOT01A`, `NRF52840_DK`
+    - The supported toolchains are listed [in the build tools overview](../build-tools/index.html).
+
+    Example for FRDM-K64F and GCC:
+
+    ```
+    mbed-tools configure -m K64F -t GCC_ARM
+    mbed_config.cmake has been generated and written to '/Users/UserName/Development/Blinky/.mbedbuild'
+    ```
+
+## Build the project
+
+Use CMake to build your application:
+
+1. Navigate to the project's root folder.
+1. Set the build parameters:
+
+    ```
+    cmake -S . -B cmake_build -GNinja -DCMAKE_BUILD_TYPE=<profile>
+    ```
+    - -S <path-to-source>: Path to the root directory of the CMake project. We use `.` to indicate we're building from the current directory.<!--at no point until now did we tell them to navigate to the directory, though-->
+    - -B <path-to-build>: Path to the build output directory. If the directory doesn't already exist, CMake will create it. We use `cmake_build` as the output directory name; you can use a different name.
+    - -GNinja: To use the Ninja tool.
+    - -DCMAKE_BUILD_TYPE: Build type. The value (`profile`) can be `release`, `debug` or `develop`, as [explained in program setup](../program-setup/build-profiles-and-rules.html).
+
+1. Build:
+
+    ```
+    cmake --build cmake_build
+    ```
+
+    This generates two files: BIN and HEX in the build output directory (`cmake_build` in this example).
```

Comment:
```suggestion
    This generates a HEX file in the build output directory (`cmake_build` in this example).
```

Reply:
It generates both `HEX` and `bin` files, which can be used.

Reply:
But when I asked how you know which one to use, the answer I got was "it's HEX".

Reply:
The build will generate both hex and bin. It’s up to the developer to use either of them. But the recommendation is the `HEX` file.

Reply:
What's the advantage of the HEX file?

Reply:
The advantage with hex is specifying the address for the image. During flashing only the address range mentioned in the `hex` are modified.

Reply:
That actually sounds like a pretty big deal that we should maybe consider focusing on. Maybe not right in the middle of the build process, though. Maybe the build tools intro? Do they all do this?